content/en/blog/2022/cni-mode/index.md

```diff
@@ -0,0 +1,69 @@
+<!--有了当前 Pod IP 之后，我们可以根据这个 Pod 的配置，确认流量处理路径（比如 `excludeOutboundPorts`）。-->
+With the current Pod IP, we can confirm the traffic processing path (such as `excludeOutboundPorts`) according to the configuration of this Pod.
+<!--同时，我们还使用 Pod 优化了之前解决四元组冲突的方案，改为使用 `bpf_bind` 绑定源 IP，目的 IP 直接使用 `127.0.0.1`，为了后续支持 IPv6 做准备。-->
+In addition, we also optimized the previous solution to quadruple conflicts by using `bpf_bind` to bind the source IP and using `127.0.0.1` as the destination IP, preparing for future support for IPv6.
+<!--### 如何处理入口流量？-->
+### How to handle ingress traffic?
+<!--为了能够处理入口流量，我们引入了 XDP 程序，XDP 程序作用在网卡上，能够对原始数据包做修改。-->
+In order to handle ingress traffic, we introduced the XDP program, which works on the network card and can modify the original data packets.
+<!--我们借助 XDP 程序，在流量到达 Pod 的时候，修改目的端口为 15006 以完成流量转发。-->
+We use the XDP program to modify the destination port as `15006` when the traffic reaches the Pod, so as to complete traffic forwarding.
+<!--同时考虑到可能存在主机直接访问 Pod 的情况，也为了减小影响范围，我们选择将 XDP 程序附加到 Pod 的网卡上。这就需要借助 CNI 的能力，在创建 Pod 时进行附加操作。-->
+Considering the possibility of direct host-pod communication, and in order to narrow down the affected scope, we attached the XDP program to the Pod's network card. This requires the ability of CNI to perform additional operations when creating Pods.
+<!--## 如何体验 CNI 模式？-->
+## How to use CNI mode?
+<!--CNI 模式默认被关闭，需要手动开启。-->
+CNI mode is disabled by default. You need to enable it manually with the following command.
+<!--可以使用以下命令一键开启：-->
+```bash
+curl -sSL https://raw.githubusercontent.com/merbridge/merbridge/main/deploy/all-in-one.yaml | sed 's/--cni-mode=false/--cni-mode=true/g' | kubectl apply -f -
+```
+
+<!--## 注意事项-->
+## Notes
+<!--### CNI 模式处于测试阶段-->
+### CNI mode is in beta
+<!--CNI 模式刚被设计和开发出来，可能存在不少问题，我们欢迎大家在测试阶段进行反馈，或者提出更好的建议，以帮助我们改进 Merbridge！-->
+The CNI mode is a new feature that may not be perfect. Anyone is welcome and appreciated to give feedback or suggestions to help improve this mode and Merbridge.
+<!--如果需要使用注入 Istio perf benchmark 等工具进行测试性能，请开启 CNI 模式，否则会导致性能测试结果不准确。-->
+In performance tests using `Istio perf benchmark` and other tools, it is suggested to enable the CNI mode, otherwise the test results may not be accurate.
+<!--### 需要注意主机是否可开启 hardware-checksum 能力-->
+### Check whether the host can enable the hardware-checksum capability
+<!--为了保证 CNI 模式的正常运行，我们默认关闭了 hardware-checksum 能力，这可能会影响到网络性能。建议大家在开启 CNI 模式前，先确认主机是否可开启 hardware-checksum 能力。如果可以开启，建议设置 `--hardware-checksum=true` 以获得最佳的性能表现。-->
+In order to ensure normal functioning of the CNI mode, the hardware-checksum capability is disabled by default, which may affect network performance. It is recommended to check whether the host can enable this before enabling the CNI mode. If yes, we suggest to set `--hardware-checksum=true` for best performance.
```

Comment:
In order to ensure normal functioning of the CNI mode, the `hardware-checksum` capability is disabled by default, which may affect network performance. It is recommended to check whether you can enable this capability on the host before enabling the CNI mode. If yes, we suggest to set `--hardware-checksum=true` for best performance.